src/v/cluster/partition.cc

```diff
@@ -504,23 +504,25 @@ partition::timequery(storage::timequery_config cfg) {
         co_return co_await cloud_storage_timequery(cfg);
     }
 
+    const bool may_answer_from_cloud
```

Comment:
> Altered timequery logic in partition.cc

The commit message needs to explain what is happening. What is the problem, what is the fix, and any other salient bits to explain the change and provide context for reviewer.

Reply:
+1

I'm sure there are plenty of resources, but here's one that I was given in my youth: https://cbea.ms/git-commit/

The stylistic things are a bit on the pedantic side and I sometimes ignore, but I particularly appreciate number 7

Reply:
There is also a little bit written in the contributing guide https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#commit-history

Reply:
Thanks for the sources, well noted. Amended commit.